packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -866,10 +870,26 @@ class _InsightsHandler {
             browser: this._platformMeta?.browserName,
             browser_version: this._platformMeta?.browserVersion,
             platform: this._platformMeta?.platformName,
-            product: this._platformMeta?.product
+            product: this._platformMeta?.product,
+            platform_version: this.getPlatformVersion()
         }
     }
 
+    private getPlatformVersion() {
+        const caps = (this._userCaps as WebdriverIO.Capabilities)
+        const bstackOptions = (this._userCaps as WebdriverIO.Capabilities)?.['bstack:options']
+        const keys = ['platformVersion', 'platform_version', 'osVersion', 'os_version']
+
+        for (const key of keys) {
+            if ( bstackOptions && bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]) {
+                return bstackOptions?.[key as keyof Capabilities.BrowserStackCapabilities]
+            } else if ((caps as WebdriverIO.Capabilities)[key as keyof WebdriverIO.Capabilities]) {
+                return (caps as WebdriverIO.Capabilities)[key as keyof WebdriverIO.Capabilities]
```

Comment:
these have been already casted above
```suggestion
                return caps[key as keyof WebdriverIO.Capabilities]
```